python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -236,12 +238,37 @@ def optimize(self, model: Model,
                 def func_test(model, sample):
                     model(sample)
                 try:
-                    result_map[method]["latency"], status =\
-                        throughput_calculate_helper(latency_sample_num, baseline_time,
-                                                    func_test, acce_model, input_sample)
-                    if status is False and method != "original":
-                        result_map[method]["status"] = "early stopped"
-                        continue
+                    if method == "original":
+                        # for original keras model, as tf.config.threading can't set thread
+                        # during running, so here we use subprocess to calculate throughput
+                        params = {"iterrun": latency_sample_num,
+                                  "func": func_test,
+                                  "model": acce_model,
+                                  "input_sample": input_sample}
+                        with tempfile.TemporaryDirectory() as temp_dir:
+                            _filename = os.path.join(temp_dir, "params")
+                            cloudpickle.dump(params, open(_filename, "wb"))
+                            if thread_num is not None:
+                                my_env = os.environ.copy()
+                                my_env["OMP_NUM_THREADS"] = str(thread_num)
+                            else:
```

Comment:
Can we calculate throughput in current process if user does not specify `thread_num` ?